volume/store/store.go

```diff
@@ -47,39 +60,94 @@ func (s *VolumeStore) AddAll(vols []volume.Volume) {
 func (s *VolumeStore) Create(name, driverName string, opts map[string]string) (volume.Volume, error) {
 	s.mu.Lock()
 	if vc, exists := s.vols[name]; exists {
-		v := vc.Volume
 		s.mu.Unlock()
-		return v, nil
+		<-vc.waitCreate
+		if vc.Volume != nil && (vc.Volume.DriverName() == driverName || driverName == volume.DefaultDriverName) {
```

Comment:
@coolljt0725 This means that if the passed in driver name is the default driver name, but the one that we already have is for some other driver (like `convoy`), then we will assume that the user wanted convoy and not something from the default driver.
This is really a problem due to `--volume-driver` on `docker run`.